Library/Homebrew/cask/cmd/info.rb

```diff
@@ -25,52 +26,70 @@ def self.help
         "displays information about the given Cask"
       end
 
+      def self.get_info(cask)
+        output = title_info(cask) + "\n"
+        output << Formatter.url(cask.homepage) + "\n" if cask.homepage
+        output << installation_info(cask)
+        repo = repo_info(cask)
+        output << repo unless repo.nil?
+        output << name_info(cask)
+        language = language_info(cask)
+        output << language unless language.nil?
+        output << artifact_info(cask)
+        caveats = Installer.print_caveats(cask)
+        output << caveats unless caveats.nil?
+        output
+      end
+
       def self.info(cask)
-        title_info(cask)
-        puts Formatter.url(cask.homepage) if cask.homepage
-        installation_info(cask)
-        repo_info(cask)
-        name_info(cask)
-        language_info(cask)
-        artifact_info(cask)
-        Installer.print_caveats(cask)
+        puts get_info(cask)
       end
 
       def self.title_info(cask)
         title = "#{cask.token}: #{cask.version}"
         title += " (auto_updates)" if cask.auto_updates
-        puts title
+        title
       end
 
       def self.formatted_url(url)
         "#{Tty.underline}#{url}#{Tty.reset}"
       end
 
       def self.installation_info(cask)
+        install_info = ""
         if cask.installed?
           cask.versions.each do |version|
             versioned_staged_path = cask.caskroom_path.join(version)
-
-            puts versioned_staged_path.to_s
-              .concat(" (")
-              .concat(versioned_staged_path.exist? ? versioned_staged_path.abv : Formatter.error("does not exist"))
-                                      .concat(")")
+            install_info << versioned_staged_path.to_s
+                            .concat(" (")
+                            .concat(
+                              if versioned_staged_path.exist?
+                              then versioned_staged_path.abv
+                              else Formatter.error("does not exist")
+                              end,
+                            ).concat(")")
+            install_info << "\n"
           end
+          install_info
         else
-          puts "Not installed"
+          "Not installed\n"
```

Comment:
Another bikeshed but it feels strange to have a function returning strings with newlines in them.

Reply:
I'll try to move newlines out of the helper methods and into `get_info` where I can; In this case if I removed this newline I would have to strip the newline from `install_info` when it gets returned on line 73, because it gets appended in a loop on line 71. Stripping a newline in a callee method only to re-append it in the caller also feels funky to me.